src/App.tsx

```diff
@@ -15,81 +16,132 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
 enum SortType {
   NONE,
   ALPHABET,
   LENGTH,
 }
 
-type ReorderOptions = {
-  sortType: SortType,
-  isReversed: boolean,
-};
-
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+): string[] {
+  let visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+    default:
+      visibleGoods = [...goodsFromServer];
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: number,
+};
+
+export class App extends React.PureComponent<{}, State> {
+  goods = [...goodsFromServer];
+
+  state = {
+    isReversed: false,
+    sortType: 0,
+  };
+
+  buttonClickReverse = (): void => {
+    this.setState(prevState => (
+      { ...prevState, isReversed: !prevState.isReversed }));
+  };
+
+  buttonClickByAlphabet = (): void => {
+    this.setState(prevState => ({ ...prevState, sortType: SortType.ALPHABET }));
```

Comment:
why do you need prevState ?

Reply:
You can have more conditions in future and this make sense.

Reply:
You're not using prevState, why do you need functional setState? This task hasn't future

Reply:
still not resolved